### PR TITLE
Delete glossary Dynamic programming language

### LINKED ISSUE #10616: Issue with "Dynamic programming language": (short summary here please)
MDN URL: https://developer.mozilla.org/en-US/docs/Glossary/Dynamic_programming_language

#### What information was incorrect, unhelpful, or incomplete?
First sentence doesn't totally make sense. Get confused near "operations otherwise".

#### Specific section or headline?
In first section, first line of  'Dynamic programming language'.

#### What did you expect to see?
First sentence to be more clear.

#### Did you test this? If so, how?
One of my friend also said this sentence to be ugly.

image

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/glossary/dynamic_programming_language`
* MDN URL: https://developer.mozilla.org/en-US/docs/Glossary/Dynamic_programming_language
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/glossary/dynamic_programming_language/index.md
* Last commit: https://github.com/mdn/content/commit/4a9f9d2e9c1811ce3519a7061e906db9200a5e97
* Document last modified: 2021-10-08T00:24:32.000Z

</details>

### PR DESCRIPTION
**Pull request by hamishwillee**
Related to #10616

EDITED. This deletes the page and the case where it was used.

### CODE REVIEW

**Comment by himanshugarg** [comment]
@hamishwillee please allow me to nitpick in case it helps you see an alternate point of view:-

> A **dynamic programming language** is a programming language in which some [**operations are performed during execution**] [**that would be performed during compilation in a static  programming language**].

can also be read as 

> A **dynamic programming language** is a programming language in which some operations are performed during [**execution that would be performed during compilation in a static  programming language**].

> For example, a static programming language would only allow plugins to be added during compilation, while a dynamic programming language _might_ allow plugins to be loaded at runtime.

But then, the editors Eclipse, Notepad++ also allow plugins to be added. They are Java/C++ based.

> its behavior can be fully understood using static analysis. By contrast the variables and behaviour of a dynamic language depends on the environment at runtime.

I think we could qualify the word `behavior` but with what, I don't know. The behavior is also affected by the inputs and the environment in which the program runs. And there is a typo in one of the words I am not sure which.

**Comment by hamishwillee** [comment]
@himanshugarg Thanks. I _think_ this is all technically correct, but unfortunately if you can be confused by it, then it isn't good enough.

>>For example, a static programming language would only allow plugins to be added during compilation, while a dynamic programming language might allow plugins to be loaded at runtime.

> But then, the editors Eclipse, Notepad++ also allow plugins to be added. They are Java/C++ based.

Yes, and that is correct and consistent with the statement. 
- Eclipse and Java are based on dynamic languages. They allow plugins to be loaded at runtime.
- If they were static languages you could not add plugins at runtime - only at compile time.
- Perhaps you are thinking about the fact that C is a _statically typed_ language? If so, that has nothing to do with this topic. 

The word "behaviors" is supposed to captures that this isn't just plugin loading - it could be a bunch of things - like the ability to create new objects on the fly. You can do the same things in both, it's more a matter of how hard they are to do, and when you can do them. 

With respect to inputs and the way a program runs that is tricky. As I understand the difference here, it is mostly about variables vs values. For a static program all the variables that can be created and the memory that can be allocated is determined at compile time - of course the values might change at runtime. For a dynamic language the you can even define new data structures on the fly. 

Perhaps I don't understand it well enough :-( I'll take another try.

**Comment by hamishwillee** [comment]
OK @himanshugarg I reverted most of my changes so that now this just fixes the one part that I am confident of. 

This term really isn't one that has much use in this domain, so even though I am quite confident our definition is poor, I don't think that putting in time to improve it is value for money.

**Comment by hamishwillee** [comment]
@ddbeck can you look at this at some point please?

**Comment by himanshugarg** [comment]
> OK @himanshugarg I reverted most of my changes so that now this just fixes the one part that I am confident of.

Sorry for the silence after the noise. Thanks for the patience with both.

> For example, C++ and JavaScript are both dynamic programming languages, but C++ is statically typed, while Javascript is dynamically typed.

As you have noted, I was not aware of this distinction. It is definitely helpful.

**Comment by ddbeck** [comment]
OK, you might hate my suggestion here, but what do you think of just getting rid of this glossary entry?

It's used in [only one place](https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/JavaScript_basics#what_is_javascript) where the detail of what _kind_ of programming language is not terribly interesting. I'd expect that someone asking the question "What is JS?" is a beginner who isn't likely to know or care about the distinction.

If this were my PR, I'd delete the glossary entry and replace the glossary macro call with something like "JavaScript is a programming language that runs in the browser that can add interactivity to a website."

**Comment by wbamberg** [comment]
> OK, you might hate my suggestion here, but what do you think of just getting rid of this glossary entry?
> 
> It's used in [only one place](https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/JavaScript_basics#what_is_javascript) where the detail of what _kind_ of programming language is not terribly interesting. I'd expect that someone asking the question "What is JS?" is a beginner who isn't likely to know or care about the distinction.
> 
> If this were my PR, I'd delete the glossary entry and replace the glossary macro call with something like "JavaScript is a programming language that runs in the browser that can add interactivity to a website."

I agree with this.

I think one problem here is that we lack a clear definition of what makes a programming language dynamic. For example, I'd not include C++ as a dynamic programming language (and neither does the Wikipedia page, which is the only source cited for this entry: https://en.wikipedia.org/wiki/Dynamic_programming_language#Examples). Maybe I'm wrong, but that just makes this seem like a matter of opinion.

I think we can probably agree on the _kinds of things that are distinctive of dynamic programming languages_, like the ability to add or redefine an object's properties at runtime, but if we can't agree on a clear test of what makes a language "dynamic" then it's hard to see how this can be a helpful entry.

**Comment by hamishwillee** [comment]
No, I'm more than fine with this. I'll update.

File: files/en-us/learn/getting_started_with_the_web/javascript_basics/index.md [hunk 1]
```diff
@@ -11,13 +11,16 @@ tags:
...
 
-JavaScript is a programming language that adds interactivity to your website. This happens in games, in the behavior of responses when buttons are pressed or with data entry on forms; with dynamic styling; with animation, etc. This article helps you get started with JavaScript and furthers your understanding of what is possible.
+JavaScript is a programming language that adds interactivity to your website. This happens in games, in the behavior of responses when buttons are pressed or with data entry on forms; with dynamic styling; with animation, etc.
+This article helps you get started with JavaScript and furthers your understanding of what is possible.
 
 ## What is JavaScript?
 
-{{Glossary("JavaScript")}} ("JS" for short) is a full-fledged {{Glossary("Dynamic programming language", "dynamic programming language")}} that can add interactivity to a website. It was invented by Brendan Eich (co-founder of the Mozilla project, the Mozilla Foundation, and the Mozilla Corporation).
+{{Glossary("JavaScript")}} is a powerful programming language that can add interactivity to a website.
```
Comment (hamishwillee) [inline files/en-us/learn/getting_started_with_the_web/javascript_basics/index.md:19]: This is the only content change

**Comment by hamishwillee** [comment]
Done. Ready for re-review

**Review by wbamberg — APPROVED** [review]
👍 thanks Hamish and @himanshugarg , and thanks Daniel for speaking the unspeakable :).

File: files/en-us/learn/getting_started_with_the_web/javascript_basics/index.md (hunk 1 above)
Comment (wannurhadi) [reply files/en-us/learn/getting_started_with_the_web/javascript_basics/index.md:19]: Tq

**Pull request merged**